analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd

```diff
@@ -0,0 +1,290 @@
+`02-subset-fusion-files-LGAT.Rmd` gathers subttype associated Fusion annotation
+`03-subset-cnv-files-LGAT.Rmd` gathers subtype associated CNV annotation
+
+along with the above molecular data based annotation in this notebook we will include NF-1 cancer_predispositions from histology file as a criteria for LGG, NF1 as well
+
+### Setup
+```{r}
+library("tidyverse")
+```
+
+### Input
+```{r}
+# Look for git root folder
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+
+# clinical file
+lgat_manifest <- read_tsv(file.path(root_dir, 
+                               "data",
+                               "pbta-histologies.tsv"), 
+                     guess_max = 10000) %>%
+  filter( broad_histology == "Low-grade astrocytic tumor") %>%
+  select(Kids_First_Biospecimen_ID,
+         Kids_First_Participant_ID,
+         sample_id,
+         experimental_strategy,
+         cancer_predispositions)
+
+lgat_wgs <- lgat_manifest %>%
+  filter(experimental_strategy != "RNA-Seq") %>%
+  select( -experimental_strategy, -cancer_predispositions)
+
+lgat_rna <- lgat_manifest %>%
+  filter(experimental_strategy == "RNA-Seq") %>%
+  select( -experimental_strategy, -cancer_predispositions)
+
+all_lgat_matched <- full_join(lgat_wgs, lgat_rna,
+                         by = c("Kids_First_Participant_ID",
+                                "sample_id"),
+                         suffix = c("_DNA", "_RNA")) %>%
+  select(starts_with("Kids_First"), sample_id)
+
+# get subset folder
+subset_dir <- file.path(
+  root_dir,
+  "analyses",
+  "molecular-subtyping-LGAT",
+  "lgat-subset"
+)
+
+# result Dir
+resultDir <- file.path(
+  root_dir,
+  "analyses",
+  "molecular-subtyping-LGAT",
+  "results"
+)
+
+# SNV based annotation
+lgat_snv <- read_tsv(file.path(subset_dir,"LGAT_snv_subset.tsv")) 
+
+# CNV based annotation
+lgat_cnv <- read_tsv(file.path(subset_dir,"LGAT_cnv_subset.tsv")) %>%
+  # To-do add to CNV subset script
+  rename(Kids_First_Biospecimen_ID=lgat_wgs_biospecimen_ids)
+
+# Fusion base annotation
+lgat_fus <- read_tsv(file.path(subset_dir,"LGAT_fusion_subset.tsv")) 
+
+```
+
+
+### LGG, NF1
+This subtype is characterized by germline NF1 variants. We do not have germline data within OpenPBTA, but we do have information about cancer predispositions. This group can be annotated if NF-1 is present within the cancer_predispositions field of the pbta-histologies.tsv file. We should also annotate somatic loss of NF1 via either missense, nonsense mutation or copy loss.
+```{r}
+nf1_snv <- lgat_snv %>%
+  filter(NF1_mut=="Yes") %>%
+  mutate(molecular_subtype = "LGG, NF1") %>%
+  select("Kids_First_Biospecimen_ID","molecular_subtype") 
+
+nf1_germline <- lgat_manifest %>% 
+  # to keep non-RNA files only since we will match the RNA 
+  # as Kids_First_Biospecimen_ID_RNA column at a later step
+  filter(experimental_strategy != "RNA-Seq") %>%
+  # keeping "NF1 and NF2" in cancer_predisposition
+  filter(grepl("NF", cancer_predispositions)) %>%
```

Comment:
Are the `cancer_predispositions` values standardized in some way such that you could do exact matching for both NF1 and NF2?

Reply:
I updated to use specific values related to NF1/NF2 cancer predisposition to do exact matching